data/data-pipeline/data_pipeline/utils.py

```diff
@@ -198,6 +199,21 @@ def temp_folder_cleanup() -> None:
     remove_all_from_dir(data_path / "tmp")
 
 
+def check_first_run() -> bool:
+    """Checks if a local semaphore has been set to only run once the
+    calling function"""
```

Comment:
This statement is confusing to me; I think it's missing a phrase or verb? Specifically `set to only run once the calling function`: once the calling function has done what? or is it once _per_ calling function? or once the function has been called?